client/python/lib/ovmsclient/tfs_compat/grpc/tensors.py

```diff
@@ -63,43 +104,66 @@ def make_tensor_proto(values, dtype=None, shape=None):
         >>> tensor_proto = make_tensor_proto(data)
         >>> print(tensor_proto)
     '''
-
-    raise NotImplementedError
-
-def _as_numpy_dtype(tensor_dtype):
-    if tensor_dtype == DataType.DT_INVALID:
-        raise ValueError("Tensor data type invalid")
-    elif tensor_dtype == DataType.DT_FLOAT:
-        return np.float32
-    elif tensor_dtype == DataType.DT_DOUBLE:
-        return np.float64
-    elif tensor_dtype == DataType.DT_INT32:
-        return np.int32
-    elif tensor_dtype == DataType.DT_UINT8:
-        return np.uint8
-    elif tensor_dtype == DataType.DT_INT16:
-        return np.int16
-    elif tensor_dtype == DataType.DT_INT8:
-        return np.int8
-    elif tensor_dtype == DataType.DT_STRING:
-        return np.str
-    elif tensor_dtype == DataType.DT_COMPLEX64:
-        return np.complex64
-    elif tensor_dtype == DataType.DT_COMPLEX128:
-        return np.complex128
-    elif tensor_dtype == DataType.DT_INT64:
-        return np.int64
-    elif tensor_dtype == DataType.DT_BOOL:
-        return np.bool
-    elif tensor_dtype == DataType.DT_UINT16:
-        return np.uint16
-    elif tensor_dtype == DataType.DT_UINT32:
-        return np.uint32
-    elif tensor_dtype == DataType.DT_UINT64:
-        return np.uint64
+    if shape is None:
+        infered_shape = []
     else:
-        return np.uint32
-
+        infered_shape = shape
+    
+    value = values
+    while dtype is None or (shape is None and _is_array_like(value)):
+        if _is_array_like(value):
+            if shape is None:
+                infered_shape.append(len(value))
+            if(len(value)>0):
```

Comment:
If value contains no elements we will end up in endless loop. There is also no test for that.